pkg/sync/sync.go

```diff
@@ -186,7 +188,8 @@ func (t *Task) PurgeSnapshots(skip bool) error {
 		go func(rep *types.ControllerReplicaInfo) {
 			defer wg.Done()
 
-			repClient, err := replicaClient.NewReplicaClient(rep.Address)
+			// We don't know the replica's instanceName, so create a client without it.
+			repClient, err := replicaClient.NewReplicaClient(rep.Address, t.volumeName, "")
```

Comment:
If we do this https://github.com/longhorn/longhorn-engine/pull/902#discussion_r1266149496, we will have instance name for verification here and similar Task function in this file

Reply:
Please see my response. Let's follow up offline for further discussion as necessary.